src/translate.py

```diff
@@ -3769,59 +3783,75 @@ def read_pointer() -> Optional[Expression]:
             assert isinstance(label, str)
             return self.address_of_gsym(label)
 
-        if type.is_int() or type.is_float():
-            size_bits = type.get_size_bits()
-            if size_bits == 0:
+        def for_type(type: Type) -> Optional[str]:
+            if type.is_int() or type.is_float():
+                size_bits = type.get_size_bits()
+                if size_bits == 0:
+                    return None
+                elif size_bits is None:
+                    # Unknown size; guess 32 bits
+                    size_bits = 32
+                value = read_uint(size_bits // 8)
+                if value is not None:
+                    return Literal(value, type).format(fmt)
+
+            if type.is_pointer():
+                ptr = read_pointer()
+                if ptr is not None:
+                    return as_type(ptr, type, True).format(fmt)
+
+            if type.is_ctype():
+                # TODO: Generate initializers for structs/arrays/etc.
                 return None
-            elif size_bits is None:
-                # Unknown size; guess 32 bits
-                size_bits = 32
-            value = read_uint(size_bits // 8)
-            if value is not None:
-                return Literal(value, type).format(fmt)
-
-        if type.is_pointer():
-            ptr = read_pointer()
-            if ptr is not None:
-                return as_type(ptr, type, True).format(fmt)
-
-        if type.is_ctype():
-            # TODO: Generate initializers for structs/arrays/etc.
+
+            # Type kinds K_FN and K_VOID do not have initializers
             return None
 
-        # Type kinds K_FN and K_VOID do not have initializers
-        return None
+        if sym.array_dim is None:
+            return for_type(sym.type)
+        else:
+            elements: List[str] = []
+            for _ in range(sym.array_dim):
+                el = for_type(sym.type)
+                if el is None:
+                    return None
+                elements.append(el)
+            return f"{{{', '.join(elements)}}}"
```

Comment:
I refactored this slightly so that when CType arrays are supported, they'll use nearly the same code path.

(For now I'm not supporting CTypes at all - but that shouldn't be too bad to add in a future PR)